src/Controller/MiddlewareController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Mvc\Controller;
+
+use Zend\EventManager\EventManager;
+use Zend\Http\Request;
+use Zend\Http\Response;
+use Zend\Mvc\Exception\RuntimeException;
+use Zend\Mvc\MvcEvent;
+use Zend\Psr7Bridge\Psr7Response;
+use Zend\Psr7Bridge\Psr7ServerRequest;
+use Zend\Router\RouteMatch;
+
+/**
+ * Note: I'm a terrible person
+ *
+ * @internal don't use this in your codebase, or else @ocramius will hunt you down. This is just an internal
+ * @internal hack to make middleware trigger 'dispatch' events attached to the DispatchableInterface identifier.
+ */
+final class MiddlewareController extends AbstractController
+{
+    /**
+     * @var callable
+     */
+    private $middleware;
+
+    public function __construct(callable $middleware, EventManager $eventManager, MvcEvent $event)
```

Comment:
Just a note: on the develop branch, we're now targeting Stratigility `^2.0.1`, which means we can have:

- callable middleware
- http-interop middleware

If this patch targets master, we would need to remove the typehint for `$middleware`, and also update how middleware is invoked based on whether or not it implements `Interop\Http\ServerMiddleware\MiddlewareInterface`.